.github/workflows/deploy-preview.yml

```
@@ -0,0 +1,108 @@
+          cache: 'npm'
+          cache-dependency-path: frontend/package-lock.json
+
+      - name: Install Vercel CLI
+        run: npm install -g vercel
+
+      - name: Deploy Preview to Vercel
+        env:
+          VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
+          VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+          VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID_STAGING }}
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2086:info:2:49: Double quote to prevent globbing and word splitting

(shellcheck)